### ISSUE TITLE
`nonexistent_path_in_comment` should ignore line and column references

### ISSUE THREAD

**Issue by smoelius** [trailofbits/dylint#1609]
`nonexistent_path_in_comment`'s regex includes ':' to help it identify urls (so that it can ignore them):
https://github.com/trailofbits/dylint/blob/9f1ea864ed6b750810dd991ea2d0fcd48058ea85/examples/supplementary/nonexistent_path_in_comment/src/lib.rs#L70

However, this causes the lint to pick up line and column references:
```
warning: referenced path does not exist
   --> core/src/core.rs:252:15
    |
252 |     //    --> core/src/core.rs:233:10
    |               ^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: verify the path is correct or remove the reference
    = note: `#[warn(nonexistent_path_in_comment)]` on by default
```
The lint should detect such cases and strip off the line and column references.
